packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -262,31 +280,62 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
     }
 
     /**
-     * Find the list of editors which meet the filtering criteria.
-     * @param editors the list of editors to filter.
-     * @param searchOptions the search options to apply.
+     * Determine if the URI matches any of the patterns.
+     * @param uri the editor URI.
+     * @param patterns
      */
-    protected findMatchedEditors(editors: EditorWidget[], searchOptions: SearchInWorkspaceOptions): EditorWidget[] {
-        if (!editors.length) {
-            return [];
-        }
+    protected inPatternList(uri: URI, patterns: string[]): boolean {
+        const opts: minimatch.IOptions = { dot: true, matchBase: true };
+        return patterns.some(pattern => minimatch(uri.toString(), this.convertPatternToGlob(uri, pattern), opts));
+    }
 
+    /**
+     * Check whether the given editor meets the filtering criteria.
+     */
+    protected isEditorValid(editorWidget: EditorWidget, searchOptions: SearchInWorkspaceOptions): boolean {
         const ignoredPatterns = this.getExcludeGlobs(searchOptions.exclude);
-        editors = editors.filter(widget => !ignoredPatterns.some(pattern => minimatch(
-            widget.editor.uri.toString(),
-            this.convertPatternToGlob(this.workspaceService.getWorkspaceRootUri(widget.editor.uri), pattern),
-            { dot: true, matchBase: true })));
+        if (this.inPatternList(editorWidget.editor.uri, ignoredPatterns)) {
+            return false;
+        }
+
+        const includePatterns = searchOptions.include;
+        if (!!includePatterns?.length && !this.inPatternList(editorWidget.editor.uri, includePatterns)) {
+            return false;
+        }
 
-        // Only include widgets that in `files to include`.
-        if (searchOptions.include && searchOptions.include.length > 0) {
-            const includePatterns: string[] = searchOptions.include;
-            editors = editors.filter(widget => includePatterns.some(pattern => minimatch(
-                widget.editor.uri.toString(),
-                this.convertPatternToGlob(this.workspaceService.getWorkspaceRootUri(widget.editor.uri), pattern),
-                { dot: true, matchBase: true })));
+        return true;
+    }
+
+    /**
+     * Search the active editor only and update the tree with those results.
+     */
+    protected searchActiveEditor(activeEditor: EditorWidget, searchTerm: string, searchOptions: SearchInWorkspaceOptions): void {
+        const isHavingOutsideWorkspaceResults = () => !!this.resultTree.get(this.defaultRootName);
```

Comment:
I think we can have a better name:

```suggestion
        const includesExternalResults = () => !!this.resultTree.get(this.defaultRootName);
```
or 

```suggestion
        const includesOutsideWorkspaceResults = () => !!this.resultTree.get(this.defaultRootName);
```